shell/app/electron_main_delegate.cc

```diff
@@ -142,13 +142,16 @@ bool ElectronPathProvider(int key, base::FilePath* result) {
     case DIR_SESSION_DATA:
       // By default and for backward, equivalent to DIR_USER_DATA.
       return base::PathService::Get(chrome::DIR_USER_DATA, result);
+    case DIR_SESSION_CACHE:
+      // By default and for backward, equivalent to DIR_USER_DATA.
+      return base::PathService::Get(chrome::DIR_USER_DATA, result);
     case DIR_USER_CACHE: {
 #if BUILDFLAG(IS_POSIX)
       int parent_key = base::DIR_CACHE;
 #else
       // On Windows, there's no OS-level centralized location for caches, so
       // store the cache in the app data directory.
-      int parent_key = base::DIR_ROAMING_APP_DATA;
+      int parent_key = base::DIR_LOCAL_APP_DATA;
```

Comment:
This is still a behavior change I think?

Reply:
Have you considered the following scenario?

- An existing app is relying on `userCache` and storing significant amount of data in that cache (hundreds of MB).
- This is bad, because all that data is synchronised by Windows into the roaming profile.
- The app updates Electron to a new version, which fixed the cache location.
- The user updates their app to the new version.

What happens now:
1. Application's cache is empty, because the cache location (dir) has changed. The app has to re-download or re-calculate data for the cache.
2. The old cache stays in user's roaming data dir, both on the local machine and on Windows Domain servers.

IMO, this is not great. At all.

I am proposing to:
- Mark this change as breaking (semver-major).
- In the release notes, explain the effect of this change on existing application installations. Even better, offer a recommended solution how move the cache to the new location after the app upgrade.

Reply:
I agree this corner case might happen, I have marked this as semver/major and listed it in the breaking changes.